lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -158,149 +550,75 @@ sub job_grab {
         : $limit_attempts
       : 1;
 
-    my $worker = _validate_workerid($workerid);
-    if ($worker->job) {
-        my $job = $worker->job;
-        log_warning($worker->name . " wants to grab a new job - killing the old one: " . $job->id);
-        $job->done(result => 'incomplete');
-        $job->auto_duplicate;
+    try {
+        $worker = _validate_workerid($workerid);
+
+        if ($worker->job && $allocate) {
+            my $job = $worker->job;
+            log_warning($worker->name . " wants to grab a new job - killing the old one: " . $job->id);
+            $job->done(result => 'incomplete');
+            $job->auto_duplicate;
+        }
+        # $worker->seen(); # We do not update caps anymore on schedule. Instead, just do this on registration.
+        # # That means if a worker change caps, needs to be restarted instead of posting them always.
+        # }
```

Comment:
here i would just delete the `}` since the comment is an advice for the developer:  in the old logic we used to update worker seen, and with it the worker capabilities - now not anymore.